### PR TITLE
CLANG: update expected output for `CLANG-14 -Xanalyzer` and set clang-min-version to 14 

### PR DESCRIPTION
**Pull request by vmihalko**

### CODE REVIEW

File: tests/find-tools.cmake [hunk 1]
```diff
@@ -39,10 +39,15 @@ set(gcc_min_version 12)
...
+set(clang_min_version 14)
+
+# command that returns major version number of the specified clang executable
+set(clang_version_cmd "echo __clang_major__ | ${TOOL_EXEC_clang} -E - | tail -1")
+
 # probe for available static analyzers and formal verification tools
 append_tool_on_succ(gcc        "test ${gcc_min_version} -le \$(${gcc_version_cmd})")
-append_tool_on_succ(clang      "${TOOL_EXEC_clang} --analyze -Xanalyzer \
-                                -analyzer-output=text -xc ${empty_main}")
+append_tool_on_succ(clang        "test ${clang_min_version} -le \$(${clang_version_cmd})")
```
Comment (kdudka) [inline tests/find-tools.cmake:50]: The column is misaligned here.

**Review by kdudka — APPROVED** [review]

**Comment by lzaoral** [comment]
@vmihalko, could you try to execute the new clang version also with tests that are disabled at the moment?

**Comment by lzaoral** [comment]
@vmihalko, the commit message of 40213f5ca3211f27b4e3b970edcd9eb47ff7a6fc and its contents do something completely different.

**Comment by vmihalko** [comment]
> @vmihalko, could you try to execute the new clang version also with tests that are disabled at the moment?

no.

**Review by lzaoral — APPROVED** [review]

**Pull request merged**